app/frontend/translations.js

```diff
@@ -41,7 +41,8 @@ const translations = {
 
   CoinAmountError: () => 'CoinAmounError: Unsupported amount of coins.',
   CryptoProviderError: ({message}) => `CryptoProviderError: ${message}`,
-  NetworkError: () => 'NetworkError: nection failed. Please check your network connection.',
+  NetworkError: () =>
```

Comment:
we can do better by going back to the "request" helper: https://github.com/vacuumlabs/adalite/blob/master/app/frontend/wallet/helpers/request.js#L22 and making the returned error more granular. As you can see, we are currently returning "Network Error" regardless of the response code, but we could refactor it for example like this:

```
const request = async function request(url, method = 'GET', body = null, headers = {}) {
  let requestParams = {
    method,
    headers,
    credentials: 'include',
  }
  if (method.toUpperCase() !== 'GET') {
    requestParams = Object.assign({}, requestParams, {body})
  }

  let response = null
  try {
    response = await fetch(url, requestParams)
  } catch (e) {
    throw NamedError('NetworkError', `${method} ${url} has failed with the following error: ${e}`)
  }

  if (!response) throw NamedError('NetworkError', `No response from ${method} ${url}`)
  
  if (response.status === 429) {
    await sleep(DELAY_AFTER_TOO_MANY_REQUESTS)
    return await request(url, method, body, headers)
  }
  
  if (response.status >= 500) {
    throw NamedError(
      'ServerError',
      `${method} ${url} returns error: ${response.status} on payload: ${JSON.stringify(requestParams)}`
    )
  } else if (response.status >= 400) {
    throw NamedError(
      'NetworkError',
      `${method} ${url} returns error: ${response.status} on payload: ${JSON.stringify(requestParams)}`
    )
  }

  return response.json()
}
```

So we could return a "ServerError" if the server returns an error 500 and then add a more specific message for that. 

The messages could then look like this:
```
NetworkError: () => 'NetworkError: Request to our servers has failed. Please check your network connection and if the problem persists, contact us.',
  ServerError: () => 'ServerError: Our servers are probably down. Please try again later and if the problem persists, contact us.',
```
Which makes clear when our servers are down vs an unknown network issue which I believe makes for a better UX and hopefully less confused users contacting us.

I also noticed we are throwing some NetworkErrors in `cardano-wallet.js` - but those are essentially wrapping network requests that would fail with it anyway, so I would suggest to remove those try/catch blocks completely as they just strip away context from the errors thrown and possibly obfuscate other non-network related errors